src/wallet/rpcdump.cpp

```diff
@@ -588,9 +591,12 @@ class AddSpendingKeyToWallet : public boost::static_visitor<bool>
                 if (!m_wallet-> AddSaplingZKey(sk, addr)) {
                     throw JSONRPCError(RPC_WALLET_ERROR, "Error adding spending key to wallet");
                 }
-    
-                m_wallet->mapSaplingZKeyMetadata[addr].nCreateTime = 1;
-                
+
+                // Sapling addresses can't have been used in transactions prior to activation.
+                m_wallet->mapSaplingZKeyMetadata[ivk].nCreateTime = std::max(
+                    1, // In case a code fork sets Sapling to always be active
+                    params.vUpgrades[Consensus::UPGRADE_SAPLING].nActivationHeight);
```

Comment:
Why do we set the create time to be the activation height?

Reply:
Good catch! Pushed a fix as discussed in our pairing.